Framework/PythonInterface/mantid/plots/plotfunctions.py

```diff
@@ -110,74 +134,103 @@ def plot(workspaces, spectrum_nums=None, wksp_indices=None, errors=False,
+        if len(nums) * len(workspaces) == 1 and waterfall:
+            waterfall = False
 
-    if tiled:
-        ws_index = [(ws, index) for ws in workspaces for index in nums]
-        for index, ax in enumerate(axes):
-            if index < len(ws_index):
-                _do_single_plot(ax, [ws_index[index][0]], errors, False, [ws_index[index][1]], kw, plot_kwargs)
-            else:
-                ax.axis('off')
-    else:
-        show_title = ("on" == ConfigService.getString("plots.ShowTitle").lower()) and not overplot
-        ax = overplot if isinstance(overplot, MantidAxes) else axes[0]
-        ax.axis('on')
-        _do_single_plot(ax, workspaces, errors, show_title, nums, kw, plot_kwargs, log_name, log_values)
+        # The plot's initial xlim and ylim are used to offset each curve in a waterfall plot.
+        # Need to do this whether the current curve is a waterfall plot or not because it may be converted later.
+        if not overplot:
+            datafunctions.set_initial_dimensions(ax)
 
-    # Can't have a waterfall plot with only one line.
-    if len(nums)*len(workspaces) == 1 and waterfall:
-        waterfall = False
+        if waterfall:
+            ax.set_waterfall(True)
 
-    # The plot's initial xlim and ylim are used to offset each curve in a waterfall plot.
-    # Need to do this whether the current curve is a waterfall plot or not because it may be converted later.
-    if not overplot:
-        datafunctions.set_initial_dimensions(ax)
+        if not overplot:
+            fig.canvas.set_window_title(figure_title(workspaces, fig.number))
+        else:
+            if ax.is_waterfall():
+                for i in range(len(nums) * len(workspaces)):
+                    errorbar_cap_lines = datafunctions.remove_and_return_errorbar_cap_lines(ax)
+                    datafunctions.convert_single_line_to_waterfall(ax, len(ax.get_lines()) - (i + 1))
 
-    if waterfall:
-        ax.set_waterfall(True)
+                    if ax.waterfall_has_fill():
+                        datafunctions.waterfall_update_fill(ax)
 
-    if not overplot:
-        fig.canvas.set_window_title(figure_title(workspaces, fig.number))
-    else:
-        if ax.is_waterfall():
-            for i in range(len(nums)*len(workspaces)):
-                errorbar_cap_lines = datafunctions.remove_and_return_errorbar_cap_lines(ax)
-                datafunctions.convert_single_line_to_waterfall(ax, len(ax.get_lines()) - (i + 1))
+                    ax.lines += errorbar_cap_lines
 
-                if ax.waterfall_has_fill():
-                    datafunctions.waterfall_update_fill(ax)
+    # This updates the toolbar so the home button now takes you back to this point.
+    # The try catch is in case the manager does not have a toolbar attached.
+    try:
+        fig.canvas.manager.toolbar.update()
+    except AttributeError:
+        pass
 
-                ax.lines += errorbar_cap_lines
+    fig.canvas.draw()
```

Comment:
Do we want this to call `show`? `matplotlib` separates the plot and `draw/show`. Could you help me understand why we do this here as this is quite a low-level function.